### PR TITLE
Webserver refactor

### PR DESCRIPTION
**Pull request by Paul-Hess**

### CODE REVIEW

**Comment by Exide** [comment]
I'm going to hold off reviewing this until the conflicts are resolved.

**Comment by Paul-Hess** [comment]
this and #1086 have passed a day on DLS and are ready for review

File: src/test/javascript/integration/webhook_error_channel_spec.js [hunk 1]
```diff
@@ -109,6 +117,7 @@ describe(__filename, () => {
 
     it('waits for the webhook to give up', async () => {
         const clause = (response) => {
+            console.log('response.body', response.body.errors);
```
Comment (Exide) [inline src/test/javascript/integration/webhook_error_channel_spec.js]: Two things.

a) The string prefix should read `response.body.errors`
b) This line should be after the next line so we can use the `errors` constant that has already been safely extracted

**Comment by Exide** [comment]
Looks good

File: src/test/javascript/integration/webhook_error_channel_spec.js (hunk 1 above)
Comment (Paul-Hess) [reply src/test/javascript/integration/webhook_error_channel_spec.js]: good point

**Pull request merged**